src/components/tx/SignOrExecuteForm/index.tsx

```diff
@@ -36,218 +27,85 @@ type SignOrExecuteProps = {
+  useEffect(() => setTx(props.safeTx), [props.safeTx])
 
-  const [advancedParams, setAdvancedParams] = useAdvancedParams({
-    nonce: tx?.data.nonce,
-    gasLimit,
-    safeTxGas: tx?.data.safeTxGas,
-  })
-
-  // Check if transaction will fail
-  const { executionValidationError, isValidExecutionLoading } = useIsValidExecution(
-    willExecute ? tx : undefined,
-    advancedParams.gasLimit,
-  )
+  // Update the tx when the advancedParams change
+  useEffect(() => {
+    if (nonceReadonly || !tx?.data) return
+    if (tx.data.nonce === advancedParams.nonce && tx.data.safeTxGas === advancedParams.safeTxGas) return
 
-  // Estimating gas
-  const isEstimating = willExecute && gasLimitLoading
-  // Nonce cannot be edited if the tx is already proposed, or signed, or it's a rejection
-  const nonceReadonly = !isCreation || !!tx?.signatures.size || isRejection
-
-  // Sign transaction
-  const onSign = async (): Promise<string | undefined> => {
-    return await signTx(tx, txId, origin)
-  }
-
-  // Execute transaction
-  const onExecute = async (): Promise<string | undefined> => {
-    const txOptions = getTxOptions(advancedParams, currentChain)
-    return await executeTx(txOptions, tx, txId, origin, willRelay)
-  }
-
-  // On modal submit
-  const handleSubmit = async (e: SyntheticEvent) => {
-    e.preventDefault()
-    setIsSubmittable(false)
-    setSubmitError(undefined)
-
-    try {
-      await (willExecute ? onExecute() : onSign())
-    } catch (err) {
-      logError(Errors._804, (err as Error).message)
-      setIsSubmittable(true)
-      setSubmitError(err as Error)
-      return
-    }
-
-    onSubmit()
-  }
-
-  // On advanced params submit (nonce, gas limit, price, etc), recreate the transaction
-  const onAdvancedSubmit = async (data: AdvancedParameters) => {
-    // If nonce was edited, create a new tx with that nonce
-    if (tx && (data.nonce !== tx.data.nonce || data.safeTxGas !== tx.data.safeTxGas)) {
-      try {
-        setTx(await createTx({ ...tx.data, safeTxGas: data.safeTxGas }, data.nonce))
-      } catch (err) {
-        logError(Errors._103, (err as Error).message)
-        return
-      }
-    }
-
-    setAdvancedParams(data)
-  }
-
-  const cannotPropose = !isOwner && !onlyExecute // Can't sign or create a tx if not an owner
-  const submitDisabled =
-    !isSubmittable ||
-    isEstimating ||
-    !tx ||
-    disableSubmit ||
-    cannotPropose ||
-    isValidExecutionLoading ||
-    (willExecute && isExecutionLoop)
-
-  const error = props.error || (willExecute ? gasLimitError || executionValidationError : undefined)
+    createTx({ ...tx.data, safeTxGas: advancedParams.safeTxGas }, advancedParams.nonce)
+      .then(setTx)
+      .catch((err) => logError(Errors._103, (err as Error).message))
+  }, [nonceReadonly, tx?.data, advancedParams.nonce, advancedParams.safeTxGas])
 
   return (
-    <form onSubmit={handleSubmit}>
```

Comment:
By removing the `form` element here we lost some padding. We could add a div instead or adjust the css for DialogContent

Reply:
Why does it matter? There'll be a new design for all of these things.